packages/kbn-content-management/table_list/src/table_list_view/table_list_view.tsx

```diff
@@ -0,0 +1,470 @@
+    }
+
+    // Add "Actions" column
+    if (editItem) {
+      const actions: EuiTableActionsColumnType<T>['actions'] = [
+        {
+          name: (item) => {
+            return i18n.translate('contentManagement.tableList.listing.table.editActionName', {
+              defaultMessage: 'Edit {itemDescription}',
+              values: {
+                itemDescription: get(item, 'attributes.title'),
+              },
+            });
+          },
+          description: i18n.translate(
+            'contentManagement.tableList.listing.table.editActionDescription',
+            {
+              defaultMessage: 'Edit',
+            }
+          ),
+          icon: 'pencil',
+          type: 'icon',
+          enabled: (v) => !(v as unknown as { error: string })?.error,
+          onClick: editItem,
+        },
+      ];
+
+      columns.push({
+        name: i18n.translate('contentManagement.tableList.listing.table.actionTitle', {
+          defaultMessage: 'Actions',
+        }),
+        width: '100px',
+        actions,
+      });
+    }
+
+    return columns;
+  }, [stateTableColumns, customTableColumn, getTagsColumnDefinition, editItem]);
+  const itemsById = useMemo(() => {
+    return keyBy(items, 'id');
+  }, [items]);
+  const selectedItems = useMemo(() => {
+    return selectedIds.map((id) => itemsById[id]);
+  }, [selectedIds, itemsById]);
+
+  // ------------
+  // Callbacks
+  // ------------
+  const fetchItems = useCallback(async () => {
+    dispatch({ type: 'onFetchItems' });
+
+    try {
+      const idx = ++fetchIdx.current;
+
+      const { searchQuery: searchQueryParsed, references } = searchQueryParser
+        ? searchQueryParser(searchQuery)
+        : { searchQuery, references: undefined };
+
+      const response = await findItems(searchQueryParsed, references);
+
+      if (!isMounted.current) {
+        return;
+      }
+
+      if (idx === fetchIdx.current) {
+        dispatch({
+          type: 'onFetchItemsSuccess',
+          data: {
+            response,
+          },
+        });
+      }
+    } catch (err) {
+      dispatch({
+        type: 'onFetchItemsError',
+        data: err,
+      });
+    }
+  }, [searchQueryParser, searchQuery, findItems]);
+
+  const deleteSelectedItems = useCallback(async () => {
+    if (isDeletingItems) {
+      return;
+    }
+
+    dispatch({ type: 'onDeleteItems' });
+
+    try {
+      await deleteItems!(selectedItems);
+    } catch (error) {
+      notifyError({
+        title: (
+          <FormattedMessage
+            id="contentManagement.tableList.listing.unableToDeleteDangerMessage"
+            defaultMessage="Unable to delete {entityName}(s)"
+            values={{ entityName }}
+          />
+        ),
+        text: error,
+      });
+    }
+
+    fetchItems();
+
+    dispatch({ type: 'onItemsDeleted' });
+  }, [deleteItems, entityName, fetchItems, isDeletingItems, notifyError, selectedItems]);
+
+  const renderCreateButton = useCallback(() => {
```

Comment:
All of these `render[X]` functions need to become components.  This is why this file is so large Github won't even show it expanded by default.

Reply:
Ok. Will split up more in a future PR 👍